pkg/cmd/resources.go

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright contributors to the Galasa project
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package cmd
+
+import (
+	"github.com/spf13/cobra"
+)
+
+type ResourcesCmdValues struct {
+	bootstrap string
+	filePath  string
+}
+
+type ResourcesCommand struct {
+	cobraCommand *cobra.Command
+	values       *ResourcesCmdValues
+}
+
+// ------------------------------------------------------------------------------------------------
+// Constructors
+// ------------------------------------------------------------------------------------------------
+
+func NewResourcesCmd(rootCommand GalasaCommand) (GalasaCommand, error) {
+	cmd := new(ResourcesCommand)
+	err := cmd.init(rootCommand)
+	return cmd, err
+}
+
+// ------------------------------------------------------------------------------------------------
+// Public functions
+// ------------------------------------------------------------------------------------------------
+
+func (cmd *ResourcesCommand) Name() string {
+	return COMMAND_NAME_RESOURCES
+}
+
+func (cmd *ResourcesCommand) CobraCommand() *cobra.Command {
+	return cmd.cobraCommand
+}
+
+func (cmd *ResourcesCommand) Values() interface{} {
+	return cmd.values
+}
+
+// ------------------------------------------------------------------------------------------------
+// Private functions
+// ------------------------------------------------------------------------------------------------
+
+func (cmd *ResourcesCommand) init(rootCmd GalasaCommand) error {
+
+	var err error = nil
+
+	cmd.values = &ResourcesCmdValues{}
+	cmd.cobraCommand, err = cmd.createCobraCommand(rootCmd)
+
+	return err
+}
+
+func (cmd *ResourcesCommand) createCobraCommand(rootCommand GalasaCommand) (*cobra.Command, error) {
+
+	var err error
+
+	resourcesCobraCmd := &cobra.Command{
+		Use:   "resources",
+		Short: "Manages resources in an ecosystem",
+		Long:  "Allows interaction with the Resources endpoint to create and maintain resources in the Galasa Ecosystem",
+	}
+
+	addBootstrapFlag(resourcesCobraCmd, &cmd.values.bootstrap)
+	resourcesCobraCmd.PersistentFlags().StringVarP(&cmd.values.filePath, "file", "f", "",
+		"Fully qualified path to the file to action against the ecosystem."+
```

Comment:
The file containing yaml definitions of resources to be applied manipulated by this command. This can be a fully-qualified path or path relative to the current directory.

I am assuming it doesn't actually have to be a fully-qualified file right ? else my suggestion above isn't correct. Why not accept relative file paths ?

Reply:
The story suggested that xxx.yaml was a decent input.